Source/GSXib5KeyedUnarchiver.m

```diff
@@ -0,0 +1,3391 @@
+          // destination  and source string tags are still the same (so far) and loaded
+          // by base class...
+        }
+      else
+        {
+          [NSException raise: NSInvalidArgumentException
+                      format: @"Can't decode %@ with %@.",NSStringFromClass([self class]),
+           NSStringFromClass([coder class])];
+        }
+    }
+  return self;
+}
+
+@end
+
+@interface IBUserDefinedRuntimeAttribute5 : IBUserDefinedRuntimeAttribute
+@end
+
+
+@implementation IBUserDefinedRuntimeAttribute5
+
+- (id) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+
+  if (self)
+    {
+      if ([coder allowsKeyedCoding])
+        {
+          [self setTypeIdentifier: [coder decodeObjectForKey: @"type"]];
+
+          // Decode value properly...
+          if ([@"boolean" isEqualToString: typeIdentifier])
+            [self setValue: [NSNumber numberWithBool: ([@"YES" isEqualToString: value] ? YES : NO)]];
+          else if ([@"image" isEqualToString: typeIdentifier])
+            [self setValue: [NSImage imageNamed: value]];
+          else if ([@"number" isEqualToString: typeIdentifier])
+            [self setValue: [coder decodeObjectForKey: @"value"]];
+          else if ([@"point" isEqualToString: typeIdentifier])
+            [self setValue: [coder decodeObjectForKey: @"value"]];
+          else if ([@"size" isEqualToString: typeIdentifier])
+            [self setValue: [coder decodeObjectForKey: @"size"]];
+          else if ([@"rect" isEqualToString: typeIdentifier])
+            [self setValue: [coder decodeObjectForKey: @"value"]];
+          else if ([@"nil" isEqualToString: typeIdentifier])
+            [self setValue: nil];
+          else
+            NSWarnMLog(@"type: %@ value: %@ (%@)", typeIdentifier, value, [value class]);
+        }
+    }
+
+  return self;
+}
+
+@end
+
+
+@implementation GSXib5KeyedUnarchiver
+
+static NSDictionary *XmltagToObjectClassCrossReference = nil;
+static NSArray      *XmltagsNotStacked = nil;
+static NSArray      *XmltagsToSkip = nil;
+static NSArray      *ClassNamePrefixes = nil;
+static NSDictionary *XmlKeyMapTable = nil;
+static NSDictionary *XmlTagToDecoderSelectorMap = nil;
+static NSDictionary *XmlKeyToDecoderSelectorMap = nil;
+static NSArray      *XmlKeysDefined  = nil;
+static NSArray      *XmlReferenceAttributes  = nil;
+static NSArray      *XmlConnectionRecordTags  = nil;
+static NSArray      *XmlBoolDefaultYes  = nil;
+
++ (void) initialize
+{
+  if (self == [GSXib5KeyedUnarchiver class])
+    {
+      // Only check one since we're going to load all once...
+      if (XmltagToObjectClassCrossReference == nil)
+        {
+          // These define XML tags (i.e. <objects ...) that should be allocated as the
+          // associated class...
+          XmltagToObjectClassCrossReference =
+            [NSDictionary dictionaryWithObjectsAndKeys:
+                            @"NSMutableArray", @"objects",
```

Comment:
Why does this (and some of the similar code below) indent by 16 spaces? Not sure of GNU coding style off the top of my head, but going by my `$DAYJOB` thinking, four spaces should suffice for a perfectly readable codestyle.